File: scratch/gopurs-adt-reuse-validation-20260916/before-fficc/src/Test/ArrayOpsFFI.php
<?php
$exports['runArrayOpsFFI'] = function($limit) {
    $n = (int)$limit;
    
    $range = function($start) {
        return function($end) use ($start) {
            $arr = [];
            $step = $end >= $start ? 1 : -1;
            for ($i = $start; $step > 0 ? $i <= $end : $i >= $end; $i += $step) {
                $arr[] = $i;
            }
            return $arr;
        };
    };

    $filter = function($p) {
        return function($arr) use ($p) {
            $res = [];
            foreach ($arr as $x) {
                if ($p($x)) {
                    $res[] = $x;
                }
            }
            return $res;
        };
    };

    $sumArray = function($arr) {
        $sum = 0;
        foreach ($arr as $x) {
            $sum += $x;
        }
        return $sum;
    };

    $rng = $range(1)($n);
    $filtered = $filter(function($x) { return $x % 2 === 0; })($rng);
    return $sumArray($filtered);
};
return $exports;

File: scratch/compact-guards-integration-20260916/after/src/Test/ArrayOpsFFI.rs
pub fn Test_ArrayOpsFFI_runArrayOpsFFI(mut limit: i64) -> i64 {
    let mut arr = Vec::new();
    let step = if limit >= 1 { 1 } else { -1 };
    let mut i = 1;
    loop {
        arr.push(i);
        if i == limit {
            break;
        }
        i += step;
    }
    let mut evens = Vec::new();
    for x in arr.iter() {
        if *x % 2 == 0 {
            evens.push(*x);
        }
    }
    let mut sum = 0;
    for x in evens.iter() {
        sum += *x;
    }
    sum
}
